yarn-project/pxe/src/pxe_oracle_interface/index.ts

```diff
@@ -42,31 +42,41 @@ import { MerkleTreeId, type NullifierMembershipWitness, PublicDataWitness } from
 import type { BlockHeader } from '@aztec/stdlib/tx';
 import { TxHash } from '@aztec/stdlib/tx';
 
-import { ContractDataProvider } from '../contract_data_provider/index.js';
-import type { PxeDatabase } from '../database/index.js';
-import { NoteDao } from '../database/note_dao.js';
 import { getOrderedNoteItems } from '../note_decryption_utils/add_public_values_to_payload.js';
+import type { AddressDataProvider } from '../storage/address_data_provider/address_data_provider.js';
+import type { AuthWitnessDataProvider } from '../storage/auth_witness_data_provider/auth_witness_data_provider.js';
+import type { CapsuleDataProvider } from '../storage/capsule_data_provider/capsule_data_provider.js';
+import type { ContractDataProvider } from '../storage/contract_data_provider/contract_data_provider.js';
+import { NoteDao } from '../storage/note_data_provider/note_dao.js';
+import type { NoteDataProvider } from '../storage/note_data_provider/note_data_provider.js';
+import type { SyncDataProvider } from '../storage/sync_data_provider/sync_data_provider.js';
+import type { TaggingDataProvider } from '../storage/tagging_data_provider/tagging_data_provider.js';
 import { WINDOW_HALF_SIZE, getIndexedTaggingSecretsForTheWindow, getInitialIndexesMap } from './tagging_utils.js';
 
 /**
  * A data layer that provides and stores information needed for simulating/proving a transaction.
  */
-export class PXEDataProvider implements ExecutionDataProvider {
+export class PXEOracleInterface implements ExecutionDataProvider {
   constructor(
-    private db: PxeDatabase,
-    private keyStore: KeyStore,
     private aztecNode: AztecNode,
+    private keyStore: KeyStore,
     private simulationProvider: SimulationProvider,
     private contractDataProvider: ContractDataProvider,
-    private log = createLogger('pxe:pxe_data_provider'),
+    private noteDataProvider: NoteDataProvider,
+    private capsuleDataProvider: CapsuleDataProvider,
+    private syncDataProvider: SyncDataProvider,
+    private taggingDataProvider: TaggingDataProvider,
+    private addressDataProvider: AddressDataProvider,
+    private authWitnessDataProvider: AuthWitnessDataProvider,
+    private log = createLogger('pxe:pxe_data_manager'),
```

Comment:
This means that the 'storage' box in your diagram is conceptual but doesn't exist as a component no? We access each store manually

Reply:
Yes. It's inconvenient, but I think saner given the size the DB was getting to